Further adoption of the new CoreIPC serialization mechanism

gavin-apple opened the pull request:
#### 259c58ed79ec59a15a4e2bf5b8f5bbb5b7bfa447
<pre>
Further adoption of the new CoreIPC serialization mechanism
<a href="https://bugs.webkit.org/show_bug.cgi?id=247422">https://bugs.webkit.org/show_bug.cgi?id=247422</a>
rdar://101921684

Further porting of WebCoreArgumentCoders over to the new serialization
form. This change includes:
        - DiagnosticLoggingDomain
        - BackForwardItemIdentifier
        - PlatformColorSpace
        - DestinationColorSpace
        - WindowFeatures
        - CompositionUnderlineColor
        - CompositionUnderline
        - MediaCaptureType
        - FileChooserSettings
        - GrammarDetail
        - TextCheckingType
        - TextCheckingResult
        - TextCheckingProcessType
        - TextCheckingRequestData

Reviewed by Alex Christensen.

* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::replaceSelectionWithFragment):
(WebCore::Editor::markAllMisspellingsAndBadGrammarInRanges):
* Source/WebCore/history/BackForwardItemIdentifier.h:
(WebCore::operator!=):
(): Deleted.
(WebCore::BackForwardItemIdentifier::encode const): Deleted.
(WebCore::BackForwardItemIdentifier::decode): Deleted.
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::mediaCaptureType const):
* Source/WebCore/page/DiagnosticLoggingDomain.h:
* Source/WebCore/platform/FileChooser.h:
* Source/WebCore/platform/graphics/DestinationColorSpace.h:
(WebCore::DestinationColorSpace::encode const): Deleted.
(WebCore::DestinationColorSpace::decode): Deleted.
* Source/WebCore/platform/graphics/PlatformColorSpace.h:
* Source/WebCore/platform/text/TextChecking.h:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
* Source/WebKit/Shared/ModelIdentifier.h:
(WebKit::ModelIdentifier::encode const): Deleted.
(WebKit::ModelIdentifier::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;WindowFeatures&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WindowFeatures&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;CompositionUnderline&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;CompositionUnderline&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;FileChooserSettings&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;FileChooserSettings&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;GrammarDetail&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;GrammarDetail&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;TextCheckingRequestData&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;TextCheckingRequestData&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;TextCheckingResult&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;TextCheckingResult&gt;::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm:
(cameraDeviceForMediaCaptureType):
(-[WKFileUploadPanel presentWithParameters:resultListener:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/ios/WebCoreSupport/WebChromeClientIOS.mm:
(webMediaCaptureType):

Canonical link: <a href="https://commits.webkit.org/256485@main">https://commits.webkit.org/256485@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/77258b850713e2cbd09119e526ca2f36cfa7c1a9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95919 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5168 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28962 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105479 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/165798 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99899 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5269 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33929 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88297 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/101307 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101581 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/3880 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82526 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30917 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/99505 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85734 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87641 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73750 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39660 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19174 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37340 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20510 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/41792 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43129 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2159 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/43717 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39766 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/platform/text/TextChecking.h:
```diff
@@ -57,7 +57,7 @@ typedef uint64_t NSTextCheckingTypes;
 #endif
 
-enum TextCheckingProcessType {
+enum class TextCheckingProcessType : uint8_t {
```
`uint8_t` -> `bool`

gavin-apple replied:
Done

achristensen07 commented on Source/WebCore/history/BackForwardItemIdentifier.h:
```diff
@@ -34,14 +34,11 @@ namespace WebCore {
 struct BackForwardItemIdentifier {
     ProcessIdentifier processIdentifier;
-    enum ItemIdentifierType { };
+    enum class ItemIdentifierType : uint8_t { };
```
This is suspicious.  I think BackForwardItemIdentifier should probably just be changed to use ProcessQualified.  ObjectIdentifiers are exposed through serialized_identifiers in messages.py

achristensen07 commented on Source/WebKit/Scripts/generate-serializers.py:
```diff
@@ -479,17 +479,20 @@ def generate_impl(serialized_types, serialized_enums, headers):
-            result.append('    default:')
-            result.append('        return false;')
-            result.append('    }')
+            if len(enum.valid_values) == 0:
```
This shouldn't be necessary.  We shouldn't be serializing enums with 0 valid values.

gavin-apple committed:
Further adoption of the new CoreIPC serialization mechanism
https://bugs.webkit.org/show_bug.cgi?id=247422
rdar://101921684

Further porting of WebCoreArgumentCoders over to the new serialization
form. This change includes:
        - DiagnosticLoggingDomain
        - BackForwardItemIdentifier
        - PlatformColorSpace
        - DestinationColorSpace
        - WindowFeatures
        - CompositionUnderlineColor
        - CompositionUnderline
        - MediaCaptureType
        - FileChooserSettings
        - GrammarDetail
        - TextCheckingType
        - TextCheckingResult
        - TextCheckingProcessType
        - TextCheckingRequestData

Reviewed by Alex Christensen.

* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::replaceSelectionWithFragment):
(WebCore::Editor::markAllMisspellingsAndBadGrammarInRanges):
* Source/WebCore/history/BackForwardItemIdentifier.h:
(WebCore::operator!=):
(): Deleted.
(WebCore::BackForwardItemIdentifier::encode const): Deleted.
(WebCore::BackForwardItemIdentifier::decode): Deleted.
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::mediaCaptureType const):
* Source/WebCore/page/DiagnosticLoggingDomain.h:
* Source/WebCore/platform/FileChooser.h:
* Source/WebCore/platform/graphics/DestinationColorSpace.h:
(WebCore::DestinationColorSpace::encode const): Deleted.
(WebCore::DestinationColorSpace::decode): Deleted.
* Source/WebCore/platform/graphics/PlatformColorSpace.h:
* Source/WebCore/platform/text/TextChecking.h:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
* Source/WebKit/Shared/ModelIdentifier.h:
(WebKit::ModelIdentifier::encode const): Deleted.
(WebKit::ModelIdentifier::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder<WindowFeatures>::encode): Deleted.
(IPC::ArgumentCoder<WindowFeatures>::decode): Deleted.
(IPC::ArgumentCoder<CompositionUnderline>::encode): Deleted.
(IPC::ArgumentCoder<CompositionUnderline>::decode): Deleted.
(IPC::ArgumentCoder<FileChooserSettings>::encode): Deleted.
(IPC::ArgumentCoder<FileChooserSettings>::decode): Deleted.
(IPC::ArgumentCoder<GrammarDetail>::encode): Deleted.
(IPC::ArgumentCoder<GrammarDetail>::decode): Deleted.
(IPC::ArgumentCoder<TextCheckingRequestData>::encode): Deleted.
(IPC::ArgumentCoder<TextCheckingRequestData>::decode): Deleted.
(IPC::ArgumentCoder<TextCheckingResult>::encode): Deleted.
(IPC::ArgumentCoder<TextCheckingResult>::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm:
(cameraDeviceForMediaCaptureType):
(-[WKFileUploadPanel presentWithParameters:resultListener:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/ios/WebCoreSupport/WebChromeClientIOS.mm:
(webMediaCaptureType):

Canonical link: https://commits.webkit.org/256485@main
